lib/NetPacksLib.cpp

```diff
@@ -240,6 +240,15 @@ DLL_LINKAGE void SetAvailableHeroes::applyGs( CGameState *gs )
 	for (int i = 0; i < GameConstants::AVAILABLE_HEROES_PER_PLAYER; i++)
 	{
 		CGHeroInstance *h = (hid[i]>=0 ?  gs->hpool.heroesPool[hid[i]].get() : nullptr);
+		for (auto I = h->artifactsInBackpack.begin(), E = h->artifactsInBackpack.end();
+		     I != E; ++I)
+		{
+			if (I->artifact->artType->id == ArtifactID::GRAIL)
+			{
+				h->artifactsInBackpack.erase(I);
+				break;

```

Comment:
This will result in
```
h->artifactsInBackpack.erase(
	std::remove_if(h->artifactsInBackpack.begin(), h->artifactsInBackpack.end(),
		           [](const ArtSlotInfo& a) { return a.artifact->artType->id == ArtifactID::GRAIL; }),
	h->artifactsInBackpack.end());
```
which is neither more readable nor optimal (manual break saves useless subsequent iterations).